runner/app/live/pipelines/comfyui.py

```diff
@@ -0,0 +1,78 @@
+import os
+import json
+import torch
+from PIL import Image
+import asyncio
+import numpy as np
+
+from .interface import Pipeline
+from comfystream.client import ComfyStreamClient
+
+COMFY_UI_WORKSPACE_ENV = "COMFY_UI_WORKSPACE"
+DEFAULT_WORKFLOW_JSON = '''
+{
+    "1": {
+        "inputs": {
+            "images": ["2", 0]
+        },
+        "class_type": "SaveTensor",
+        "_meta": {
+            "title": "SaveTensor"
+        }
+    },
+    "2": {
+        "inputs": {
+            "engine": "depth_anything_vitl14-fp16.engine",
+            "images": ["3", 0]
+        },
+        "class_type": "DepthAnythingTensorrt",
+        "_meta": {
+            "title": "Depth Anything Tensorrt"
+        }
+    },
+    "3": {
+        "inputs": {},
+        "class_type": "LoadTensor",
+        "_meta": {
+            "title": "LoadTensor"
+        }
+    }
+}
+'''
+
+
+class ComfyUI(Pipeline):
+  def __init__(self, **params):
+    super().__init__(**params)
+
+    comfy_ui_workspace = os.getenv(COMFY_UI_WORKSPACE_ENV)
+    self.client = ComfyStreamClient(cwd=comfy_ui_workspace)
+
+    params = {'prompt': json.loads(DEFAULT_WORKFLOW_JSON)}
+    self.update_params(**params)
+
+    # Comfy will cache nodes that only need to be run once (i.e. a node that loads model weights)
+    # We can run the prompt once before actual inputs come in to "warmup"
+    warmup_input = torch.randn(1, 512, 512, 3)
+    asyncio.get_event_loop().run_until_complete(self.client.queue_prompt(warmup_input))
```

Comment:
- Is there a way not to use `asyncio` here? We run these pipelines on separate processes for minimum overhead, and I think ideally we should avoid these libs that alter the way the code runs as it could affect the models performance (had great performance gains by removing the pipeline logic from the main process that uses asyncio etc)
- WDYT of making a more explicit management of the event loop here, since we're not running on a process that is guaranteed to have one set up? Meaning: 
  - creating an event loop on construction with `asyncio.new_event_loop`
  - keeping a reference to it in the class and reusing it on `process_frame`
  - closing the event loop on the `__del__` function when the object gets garbage collected (apparently python does ref counting and this will happen as soon as we recreate the pipeline for example). could also do any other necessary cleanup there like closing the comfyui client

Reply:
For context, [ComfyStreamClient.queue_prompt()](https://github.com/yondonfu/comfystream/blob/34bea08e3cd16fc1e0e4cc566961df881450b1f2/src/comfystream/client.py#L21) is currently async out of convenience - it [awaits a call to the async function of ComfyUI library](https://github.com/hiddenswitch/ComfyUI/blob/44be2591df18d31e2e751e589b70f62d0317cc0b/comfy/client/embedded_comfy_client.py#L191) and a future that references the result of prompt execution. If the async version can be used for now, I think that would be the simplest path forward vs. creating a non-async variant.

@victorges IIUC if queue_prompt() is run via run_until_complete() then that just ends up mostly equivalent to calling a non-async version of queue_prompt() and while there may be some extra overhead from event loop management that should be pretty small relative to the actual computation happening in queue_prompt(). Do you have specific reason to believe that there would actually be larger perf degradation? I know you saw perf gains by moving inference code into a separate process without async code, but it is not clear to me whether the general usage of async code or specific, inefficient async code was the cause of perf degradation in that case.

Either way if the Depth-Anything-V2 workflow can run fine in real-time for now then IMO can always figure out how to further optimize separately!

Reply:
Yeah, I think that unless we see clear benefits of `asyncio.new_event_loop`, then I'd keep it as it is for now. We can optimize later. Anyway, Victor, interested in your thoughts.